### PR TITLE
Emails: Make the ThankYou react component to be a completely isolated component

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
#### Changes proposed in this Pull Request

Change a few components within the ThankYou component to be independent of the MarketPlace component.

#### Testing instructions

The test to perform for this change is the same as we had to perform in this [Pull Request](https://github.com/Automattic/wp-calypso/pull/54577):

```
There are two things we need to test:

1. The original Yoast purchase flow, and
2. The visuals for the new Professional Email page
```

The link to check the ThankYou page for the step 2 is: `/:domain/titan/setup-mailbox/thank-you/:slug`

Related to # {1200182182542585-as-1200182182542585}

### CODE REVIEW

File: client/components/thank-you/index.tsx
Comment (stephanethomas) [inline client/components/thank-you/index.tsx]: I think this felt through the cracks - there shouldn't be any closing brace here (same thing line 40).

File: client/components/thank-you/theme.ts
Comment (stephanethomas) [inline client/components/thank-you/theme.ts]: I'd like to suggest we move all type definitions in a new `types.ts` file (here [an example](https://github.com/Automattic/wp-calypso/blob/22cb668cceb98d0a166f3fe23b38ec5e1aa917dc/packages/calypso-products/src/types.ts) from a different package). This includes those two ones plus the three other ones in `index.tsx`. That way we have all the types located in a central place. This would also reduce the length of our files a bit.

File: client/components/thank-you/theme.ts
Comment (stephanethomas) [inline client/components/thank-you/theme.ts]: Why do we need to export `theme`? I think there is also no need to export `ThankYouThemeProps`.

File: client/components/thank-you/theme.ts
Comment (stephanethomas) [inline client/components/thank-you/theme.ts]: Could you add a blank line above 🙇‍♂️? We usually try to avoid making blocks of code that are too compact as they are generally harder to read. Thanks!

**Review by stephanethomas — COMMENTED** [review]
Thank you for refactoring this component! I've left a few comments that we may want to address - nothing serious though.

File: client/components/thank-you/types.ts
Comment (daledupreez) [inline client/components/thank-you/types.ts:12]: Might this be better expressed as a `TranslateResult`?

**Review by daledupreez — APPROVED** [review]
This looks good to me!
@stephanethomas, do you want another look before we merge? I _think_ all your comments were addressed.

File: client/components/thank-you/types.ts
Comment (stephanethomas) [inline client/components/thank-you/types.ts]: I still think we shouldn't export that type if it's never imported anywhere.

**Comment by stephanethomas** [comment]
I had a quick look, this is good to merge once our two minor comments have been addressed 👍.

File: client/components/thank-you/types.ts
Comment (AllTerrainDeveloper) [reply client/components/thank-you/types.ts:12]: I don't understand this change... Could you clarify?

Comment (daledupreez) [reply client/components/thank-you/types.ts:12]: Sorry! That was a bit vague. I meant that it might be clearer to indicate that `stepCta` should be a `TranslateResult` -- we're expecting a value from `translate`.
```suggestion
	stepCta: TranslateResult;
```

Comment (AllTerrainDeveloper) [reply client/components/thank-you/types.ts:12]: I don't think steCta is value from `translate` it's a _call to action_. Reviewing the code it's always a component, isn't it? I might be wrong but I can't see anywhere that this _stepCta_ is a translated string

Comment (daledupreez) [reply client/components/thank-you/types.ts:12]: You're right! No need to change this then!

**Pull request merged**
